org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java

```diff
@@ -4503,5 +4507,62 @@ public void testBug383908() throws CoreException {
 		deleteProject("P");
 	}
 }
+/*
+ * Test that we can find methods called in an anonymous type from a jar,
+ * if the primary type contains multiple anonymous types.
+ * https://github.com/eclipse-jdt/eclipse.jdt.core/issues/375
+ */
+public void testAnonymousTypeMethodReferenceJarSearch375() throws Exception {
+	String testProjectName = "gh375AnonymousTypeMethodReferenceJarSearch";
+	try {
+		IJavaProject project = setUpJavaProject(testProjectName);
+		waitForAutoBuild();
+		waitUntilIndexesReady();
 
+		JavaSearchResultCollector testResultCollector = new JavaSearchResultCollector();
+		String typeFqn = "test.gh375.TestGh375";
+		IType testType = project.findType(typeFqn);
+		assertNotNull("Failed to find test type: " + typeFqn, testType);
+		String methodName = "helloWorld";
+		IMethod testMethod = null;
+		for (IMethod method : testType.getMethods()) {
+			if (methodName.equals(method.getElementName())) {
+				testMethod = method;
+			}
+		}
+		IJavaSearchScope scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { testMethod.getAncestor(IJavaElement.PACKAGE_FRAGMENT) });
+		assertNotNull("Failed to find method: " + methodName + ", in type: " + typeFqn, testMethod);
+		int agnosticMatchRule = SearchPattern.R_EXACT_MATCH | SearchPattern.R_CASE_SENSITIVE | SearchPattern.R_ERASURE_MATCH;
+		SearchPattern pattern = SearchPattern.createPattern(testMethod, IJavaSearchConstants.REFERENCES, agnosticMatchRule);
+		SearchEngine searchEngine = new SearchEngine();
+		searchEngine.search(pattern, new SearchParticipant[] {SearchEngine.getDefaultSearchParticipant()}, scope, testResultCollector, new NullProgressMonitor());
+
+		String foundReferences = testResultCollector.toString();
+		assertFalse("Expected search to find references of method: " + testMethod, foundReferences.isEmpty());
+		List<String> results = Arrays.asList(foundReferences.split(System.lineSeparator()));
+		String[] expectedResults = {
+				"lib/testGh375.jar java.lang.Void test.gh375.<anonymous>.call()",
+		};
+		assertEquals("Unexpected search results:\n" + foundReferences, expectedResults.length, results.size());
+		Set<String> notFound = new LinkedHashSet<>();
```

Comment:
Probably can remove the loop here. I copied it from my old test, its necessary when expecting more than 1 match. In this case we have only 1 match though.